src/hal/src/command/renderpass.rs

```diff
@@ -146,10 +117,150 @@ impl<'a, B: Backend> RenderPassInlineEncoder<'a, B> {
     // TODO: set_stencil_write_mask
     // TODO: pipeline barrier (postponed)
     // TODO: begin/end query
+}}
+
+///
+pub struct RenderPassInlineEncoder<'a, B: Backend, L: Level>(pub(crate) Option<&'a mut B::CommandBuffer>, PhantomData<L>)
```

Comment:
why is there an `Option`?

Reply:
So that it's possible to move the raw buffer out of the encoder in `next_subpass_inline`/`secondary`. You can't directly remove it, even in a `self` method because of the `Drop` impl, so I just `take` it and make the `Drop` impl a no-op when there's a `None`.

Reply:
Can the secondary encoder borrow the primary encoder or it's command buffer instead of moving it in and out?

Reply:
Wouldn't that cause an issue where it would call `drop` up the tree of borrows, rather than once like it should?